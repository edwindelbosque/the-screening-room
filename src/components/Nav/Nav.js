import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import {
	setUser,
	setError,
	resetFavorites,
	resetMoviesFavorites
} from '../../actions/index';
import './Nav.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import './Nav.scss';

export const Nav = ({
	logoutCurrentUser,
	resetFavorites,
	user,
	setError,
	favorites,
	resetMoviesFavorites,
	setRandomWallpaper,
	searchResults
}) => {
	const handleLogoutClick = () => {
		logoutCurrentUser();
		resetMoviesFavorites();
		resetFavorites({});
		setError('');
	};

	const navBackdrop = {
		backgroundImage: `linear-gradient(to top, #00000000, #00000000, rgb(31, 31, 31)), url(${setRandomWallpaper})`
	};

	return (
		<div className='Nav__container' style={navBackdrop}>
			<header className='Nav'>
				<h1>The Screening Room</h1>
				<div className='Nav__ul'>
					<NavLink
						exact
						to='/'
						className='Nav__button link-wrapper'
						activeClassName='nav-active'>
						<button className='link hover-home'>Home</button>
					</NavLink>
					{searchResults.length !== 0 && (
						<NavLink
							exact
							to='/search'
							className='Nav__button link-wrapper'
							activeClassName='nav-active'>
							<button className='link hover-home'>Search</button>
						</NavLink>
					)}
					<NavLink
						to='/favorites'
						className='Nav__button link-wrapper'
						activeClassName='nav-active'>
						<button className='link hover-1'>
							Favorites
							{favorites.length ? (
								<p className='favorite-counter'>{favorites.length}</p>
							) : (
								''
							)}
						</button>
					</NavLink>
					{user.name ? (
						<NavLink
							to='/login'
							className='Nav__button link-wrapper'
							activeClassName='nav-active'>
							<button className='link hover-1' onClick={handleLogoutClick}>
								Logout
							</button>
						</NavLink>
					) : (
						<NavLink
							to='/signup'
							className='Nav__button link-wrapper'
							activeClassName='nav-active'>
							<button className='link hover-1'>Account</button>
						</NavLink>
					)}
				</div>
				{user.name && (
					<div className='welcome-banner'>
						<div className='banner-peak'></div>
						<p>
							<span>Welcome,</span> {user.name.split(' ')[0]}
							<span>!</span>
						</p>
					</div>
				)}
			</header>
		</div>
	);
};

export const mapStateToProps = ({
	movies,
	user,
	setRandomWallpaper,
	favorites,
	searchResults
}) => {
	return { movies, user, setRandomWallpaper, favorites, searchResults };
};

export const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{ setUser, setError, resetFavorites, resetMoviesFavorites },
		dispatch
	);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Nav);

Nav.propTypes = {
	logoutCurrentUser: PropTypes.func,
	user: PropTypes.object,
	setUser: PropTypes.func,
	setError: PropTypes.func,
	favorites: PropTypes.array,
	resetFavorites: PropTypes.func,
	resetMoviesFavorites: PropTypes.func
};
