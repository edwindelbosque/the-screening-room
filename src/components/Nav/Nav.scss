.Nav__container {
	height: 500px;
	background-position: center top;
	background-size: cover;
	width: 100%;
	object-fit: cover;
	color: rgb(31, 31, 31);
}

.link-wrapper {
	position: relative;
	display: block;
}

.hover-1 {
	&:after {
		content: '';
		position: absolute;
		width: 70%;
		height: 3px;
		bottom: 0;
		left: 19px;
		background-color: rgb(0, 196, 104);
		transform: scaleX(0);
		transform-origin: bottom right;
		transition: transform 0.15s;
	}

	&:hover {
		&:after {
			transform-origin: bottom left;
			transform: scaleX(1);
		}
	}
}

	.dropdown {
		position: absolute;
		top: 380px;
		left: 41.3%;
		transform: translate(-50%, -50%);
	}

	select {
		width: 380px;
		height: 50px;
		font-size: 1em;
		border: none;
		text-indent: 10px;
		font-weight: 700;
		color: rgb(255, 255, 255);
		background: linear-gradient(to right bottom, rgb(19, 206, 90), rgb(2, 146, 98),);

		&:focus {
			outline: none;
		}
	}

.hover-home {
	&:after {
		content: '';
		position: absolute;
		width: 60%;
		height: 3px;
		bottom: 0;
		left: 26px;
		background-color: rgb(0, 196, 104);
		transform: scaleX(0);
		transform-origin: bottom right;
		transition: transform 0.15s;
	}

	&:hover {
		&:after {
			transform-origin: bottom left;
			transform: scaleX(1);
		}
	}
}

.logout-button {
	color: rgba(128, 128, 128, 0.445);
	font-size: 16px;
	transition: .15s;
	
	&:hover {
		color: rgba(255, 0, 0, 0.459);
		transition: .15s;
	}
}

.Nav {
	align-items: center;
	background: transparent;
	display: flex;
	height: 75px;
	margin: auto;
	width: 93%;
	padding-right: 20px;
	justify-content: space-between;

	div {
		display: flex;
		flex-direction: row;
		height: 50px;
		list-style-type: none;
	}

	button {
		justify-content: space-between;
		width: 130px;
		text-align: center;
		height: 50px;
		cursor: pointer;
		vertical-align: middle;
		transition: 0.2s;

		&:focus {
			outline: none;
		}

		&:hover {
			color: rgb(255, 255, 255);
			transition: 0.2s;
		}
	}

	a {
		text-decoration: none;
		font-size: 1.4em;
		color: white;
		font-weight: 500;
		transition: 0.2s;
	}

	h1 {
		margin-left: 25px;
		margin-top: 80px;
		position: sticky;
		top: 0;
		font-size: 1.6em;
		width: 170px;
		color: rgb(29, 199, 134);
		background: rgba(48, 48, 48, 0.589);
		cursor: default;
		box-shadow: 0 2px 2px rgba(0, 0, 0, 0.349);
		padding: 5px 12px;
		border: 5px solid rgb(0, 196, 104);
		border-radius: 3px;
		transition: 0.2s;

		&:hover {
			color: rgb(54, 54, 54);
			background: rgb(0, 204, 109);
			border: 5px solid rgb(56, 56, 56);
			transition: 0.2s;
		}
	}
}

button {
	background-color: transparent;
	border: none;
	color: white;
	font-size: 1em;
	font-weight: 500;
	vertical-align: middle;
}

.favorite-counter {
	position: absolute;
	background-image: linear-gradient(to right bottom, #11d87b, #0cd492);
	border-radius: 20px;
	padding: 0px 6px 1px 6px;
	font-size: 16px;
	color: black;
	font-weight: 700;
	left: 105px;
	top: 0;
}

.welcome-banner {
	position: absolute;
	color: rgb(255, 255, 255);
	right: 110px;
	top: 90px;
	background-color: #018145;
	height: 25px;
	padding: 5px 15px;
	cursor: default;
	border-radius: 4px;

	.banner-peak {
		background: #018145;
		position: absolute;
		height: 30px;
		width: 30px;
		bottom: 35px;
		left: 20px;
		border-top-left-radius: 4px;
		transform: rotate(45deg);
	}

	p {
		font-weight: 600;
		font-size: 18px;
		height: 5px;
		padding-top: 10px;

		span {
			font-weight: 400;
		}
	}
}

.nav-active {
	button {
		color: rgb(0, 196, 104);
		
		&:hover {
			color: rgb(0, 196, 104);
		}

		&:active {
			color: rgb(0, 196, 104);
		}
		
		p {
			color: rgb(0, 0, 0);
		}
	}
	
	.hover-1 {
		&:after {
			content: '';
			position: absolute;
			width: 70%;
			height: 3px;
			bottom: 0;
			left: 19px;
			background-color: rgb(0, 196, 104);
			transform: scaleX(1);
			transform-origin: bottom right;
		}
	}
}

.nav-active {
	.hover-home {
		&:after {
			content: '';
			position: absolute;
			width: 60%;
			height: 3px;
			bottom: 0;
			left: 26px;
			background-color: rgb(0, 196, 104);
			transform: scaleX(1);
			transform-origin: bottom right;
		}
	}
}

@media only screen and (max-width: 820px) {
	.Nav__container {
		height: 300px;
		width: 100%;
	}

	.dropdown {
		position: absolute;
		top: 350px;
		left: 32%;
		transform: translate(-50%, -50%);
	}

	select {
		width: 380px;
		height: 50px;
		font-size: 1em;
		border: none;
		text-indent: 10px;
		font-weight: 700;
		color: rgb(255, 255, 255);
		background: linear-gradient(to right bottom, rgb(19, 206, 90), rgb(2, 146, 98),);

		&:focus {
			outline: none;
		}
	}

	.Nav {
		padding: 0;
		font-size: 90%;
		flex-direction: column;

		button {
			height: 36px;
			padding-top: 6px;
		}

		h1 {
			margin-top: 20px;
			margin-bottom: 20px;
		}
	}

	.welcome-banner {
		top: 55px;
		right: 30px;
	}
	
}

@media only screen and (max-width: 660px) {

	.welcome-banner {
		top: 140px;
		left: 50%;
		width: fit-content;
		transform: translate(-50%, -50%);
	}

	.dropdown {
		position: absolute;
		top: 410px;
		left: 25%;
		transform: translate(-50%, -50%);
	}

	select {
		width: 380px;
		height: 50px;
		font-size: 1em;
		border: none;
		text-indent: 10px;
		font-weight: 700;
		color: rgb(255, 255, 255);
		background: linear-gradient(to right bottom, rgb(19, 206, 90), rgb(2, 146, 98),);

		&:focus {
			outline: none;
		}
	}

	.Nav__container {
		height: 0;
		width: 100%;
	}

	.Nav {
		padding: 0;
		background-color: rgb(53, 53, 53);
		height: 430px;
		width: 100%;
		font-size: 90%;
		flex-direction: column;
		align-items: center;

		button {
			height: 36px;
			padding-top: 6px;
			width: 110px;
			margin-bottom: 80px;
		}

		.Nav__ul {
			margin-bottom: 185px;
		}

		h1 {
				width: 290px;
				margin: 30px 0 100px 0;
				left: 0;
		}

		a {
			font-size: 1.2em;
		}
	}

	.favorite-counter {
		position: absolute;
		background-image: linear-gradient(to right bottom, #11d87b, #0cd492);
		border-radius: 20px;
		padding: 0px 6px 1px 6px;
		font-size: 16px;
		color: black;
		font-weight: 700;
		left: 85px;
		top: 0;
	}
}

@media only screen and (max-width: 525px) {

	.welcome-banner {
		top: 140px;
		left: 50%;
		width: fit-content;
		transform: translate(-50%, -50%);
	}

	.Nav__container {
		height: 0;
		width: 100%;
	}

	.dropdown {
		position: absolute;
		top: 410px;
		right: 45%;
		transform: translate(-50%, -50%);
	}

	select {
		width: 300px;
		height: 50px;
		font-size: 1em;
		border: none;
		text-indent: 10px;
		font-weight: 700;
		color: rgb(255, 255, 255);
		background: linear-gradient(to right bottom, rgb(19, 206, 90), rgb(2, 146, 98),);

		&:focus {
			outline: none;
		}
	}

	.Nav {
		padding: 0;
		font-size: 90%;
		flex-direction: column;
		align-items: center;

		button {
			height: 36px;
			padding-top: 6px;
			width: 80px;
		}

		h1 {

				left: 0;
		}

		a {
			font-size: 1.2em;
		}
	}

	.favorite-counter {
		position: absolute;
		background-image: linear-gradient(to right bottom, #11d87b, #0cd492);
		border-radius: 20px;
		padding: 0px 6px 1px 6px;
		font-size: 16px;
		color: black;
		font-weight: 700;
		left: 60px;
		top: -10px;
	}

	.nav-active {
		
		.hover-1 {
			&:after {
				width: 87%;
				left: 5px;
			}
		}
	}

	.nav-active {
		
		.hover-home {
			&:after {
				width: 70%;
				left: 14px;
			}
		}
	}

	.hover-1 {
		&:after {
			width: 87%;
			left: 5px;
		}
	}
	
	.hover-home {
		&:after {
			width: 70%;
			left: 14px;
		}
	}
}