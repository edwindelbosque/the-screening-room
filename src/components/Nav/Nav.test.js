import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from './Nav';
import {
  setError,
  resetFavorites,
  resetMoviesFavorites,
  setSearchResults
} from '../../actions/index';

describe('Nav', () => {
  let wrapper;
  let handleCategory = jest.fn();
  let logoutCurrentUser = jest.fn();
  let searchResults = [{}, {}, {}];
  let user = {
    name: 'Vanessa Randall',
    email: 'vanessa.randall@doane.edu',
    id: 2
  };
  let favorites = [{}, {}, {}];

  beforeEach(() => {
    wrapper = shallow(
      <Nav
        logoutCurrentUser={logoutCurrentUser}
        handleCategory={handleCategory}
        searchResults={searchResults}
        user={user}
        favorites={favorites}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call logoutCurrentUser on handleLogoutClick', () => {
    let logoutCurrentUser = jest.fn();

    wrapper.find('.NavLink__button--logout').simulate('click');
    expect(logoutCurrentUser).toHaveBeenCalled();
    expect(resetMoviesFavorites).toHaveBeenCalled();
    expect(resetFavorites).toHaveBeenCalled();
    expect(setSearchResults).toHaveBeenCalled();
    expect(setError).toHaveBeenCalled();
  });
});
