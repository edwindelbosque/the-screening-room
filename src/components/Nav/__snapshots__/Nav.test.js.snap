// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<div
  className="Nav__container"
  style={
    Object {
      "backgroundImage": "linear-gradient(to top, #00000000, #00000000, rgb(31, 31, 31)), url(undefined)",
    }
  }
>
  <header
    className="Nav"
  >
    <h1>
      The Screening Room
    </h1>
    <div
      className="Nav__ul"
    >
      <NavLink
        activeClassName="nav-active"
        className="Nav__button link-wrapper dropdown"
        exact={true}
        to="/"
      >
        <button>
          <select
            onChange={[Function]}
          >
            <option
              value="playing-now"
            >
              Playing Now
            </option>
            <option
              value="top-rated"
            >
              Top Rated
            </option>
            <option
              value="trending-tv-today"
            >
              Trending TV Shows Today
            </option>
            <option
              value="trending-tv-week"
            >
              Trending TV Shows This Week
            </option>
            <option
              value="trending-movie-today"
            >
              Trending Movies Today
            </option>
            <option
              value="trending-movie-week"
            >
              Trending Movies This Week
            </option>
          </select>
        </button>
      </NavLink>
      <NavLink
        activeClassName="nav-active"
        className="Nav__button link-wrapper"
        exact={true}
        to="/search"
      >
        <button
          className="link hover-home"
        >
          Search
        </button>
      </NavLink>
      <NavLink
        activeClassName="nav-active"
        className="Nav__button link-wrapper"
        to="/favorites"
      >
        <button
          className="link hover-home"
        >
          My List
          <p
            className="favorite-counter"
          >
            3
          </p>
        </button>
      </NavLink>
      <NavLink
        activeClassName="nav-active"
        className="Nav__button link-wrapper"
        to="/login"
      >
        <button
          className="NavLink__button--logout"
          onClick={[Function]}
        >
          <span
            className="logout-button"
          >
            Logout
          </span>
        </button>
      </NavLink>
    </div>
    <div
      className="welcome-banner"
    >
      <div
        className="banner-peak"
      />
      <p>
        <span>
          Welcome,
        </span>
         
        Vanessa
        <span>
          !
        </span>
      </p>
    </div>
  </header>
</div>
`;
